node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric_test.go

```diff
@@ -552,154 +217,111 @@ func TestHelperWaitForPathToExist(t *testing.T) {
-					globReturnErr:    nil,
-				},
-			},
-
-			expErr:  nil,
-			expPath: "/dev/dm-4",
+			name:            "Should succeed",
+			cmdReturn:       []byte("Vendor Specific Identifier Extension: 0xcea5f6\n\t\t\t  [0x600fakevolumeuuid000000000111]"),
+			wwn:             "600FAKEVOLUMEUUID000000000111",
+			cmdReturnErr:    nil,
+			expErr:          nil,
+			sgInqExecutable: nil,
 		},
 	}
-
+	sgInqCmd := "sg_inq"
+	device := "/dev/dm-1"
 	for _, tc := range testCases {
 
 		t.Run(tc.name, func(t *testing.T) {
 			mockCtrl := gomock.NewController(t)
 			defer mockCtrl.Finish()
-			fake_executer := mocks.NewMockExecuterInterface(mockCtrl)
-			path := []string{getFcPath("-wwn:5"), "/dev/disk/by-path/ip-ARRAYIP-iscsi-ARRAYIQN-lun-LUNID"}
 
-			for _, dp := range path {
-				fake_executer.EXPECT().OsReadlink(dp).Return(tc.osReadLinkReturnPath, tc.osReadLinkReturnExc)
+			fake_executer := mocks.NewMockExecuterInterface(mockCtrl)
+			args := []string{"-p", "0x83", device}
+			fake_executer.EXPECT().IsExecutable(sgInqCmd).Return(tc.sgInqExecutable)
 
-				if len(tc.globReturns) == 1 {
-					fake_executer.EXPECT().FilepathGlob("/sys/block/dm-*").Return(tc.globReturns[0].globReturnfpaths, tc.globReturns[0].globReturnErr)
-				} else if len(tc.globReturns) == 2 {
-					first := fake_executer.EXPECT().FilepathGlob("/sys/block/dm-*").Return(tc.globReturns[0].globReturnfpaths, tc.globReturns[0].globReturnErr)
-					second := fake_executer.EXPECT().FilepathGlob("/sys/block/dm-4/slaves/*").Return(tc.globReturns[1].globReturnfpaths, tc.globReturns[1].globReturnErr)
+			if tc.sgInqExecutable == nil {
+				fake_executer.EXPECT().ExecuteWithTimeout(3000, sgInqCmd, args).Return(tc.cmdReturn, tc.cmdReturnErr)
+			}
 
-					gomock.InOrder(first, second)
+			helperGeneric := device_connectivity.NewOsDeviceConnectivityHelperGeneric(fake_executer)
+			wwn, err := helperGeneric.GetWwnByScsiInq(device)
+			if tc.expErr != nil || tc.expErrType != nil {
+				if err == nil {
+					t.Fatalf("Expected to fail with error, got success.")
 				}
-				helperGeneric := device_connectivity.NewOsDeviceConnectivityHelperGeneric(fake_executer)
-
-				returnPath, err := helperGeneric.GetMultipathDisk(dp)
-				if tc.expErr != nil || tc.expErrType != nil {
-					if err == nil {
-						t.Fatalf("Expected to fail with error, got success.")
+				if tc.expErrType != nil {
+					if reflect.TypeOf(err) != tc.expErrType {
+						t.Fatalf("Expected error type %v, got different error %v", tc.expErrType, reflect.TypeOf(err))
 					}
-					if tc.expErrType != nil {
-						if reflect.TypeOf(err) != tc.expErrType {
-							t.Fatalf("Expected error type %v, got different error %v", tc.expErrType, reflect.TypeOf(err))
-						}
-					} else {
-						if err.Error() != tc.expErr.Error() {
-							t.Fatalf("Expected error code %s, got %s", tc.expErr, err.Error())
-						}
+				} else {
+					if !areStringsEqualAsSet(err.Error(), tc.expErr.Error()) {
```

Comment:
do we still need `areStringsEqualAsSet`?

Reply:
sure, its good info to check what error we got and needed

Reply:
but why `AsSet`?

Reply:
I didnt write it, is this a problem?

Reply:
I don't think it is needed. do you?

Reply:
ok, done

Reply:
so is `areStringsEqualAsSet` still being used? if not, please remove it

Reply:
no, removed, done